xds/src/main/java/io/grpc/xds/WrrLocalityLoadBalancer.java

```diff
@@ -68,15 +70,32 @@ public boolean acceptResolvedAddresses(ResolvedAddresses resolvedAddresses) {
     // to produce the weighted target LB config.
     WrrLocalityConfig wrrLocalityConfig
         = (WrrLocalityConfig) resolvedAddresses.getLoadBalancingPolicyConfig();
-    Map<Locality, Integer> localityWeights = resolvedAddresses.getAttributes()
-        .get(InternalXdsAttributes.ATTR_LOCALITY_WEIGHTS);
-
-    // Not having locality weights is a misconfiguration, and we have to return with an error.
-    if (localityWeights == null) {
-      Status unavailable =
-          Status.UNAVAILABLE.withDescription("wrr_locality error: no locality weights provided");
-      helper.updateBalancingState(TRANSIENT_FAILURE, new ErrorPicker(unavailable));
-      return false;
+
+    // A map of locality weights is built up from the locality weight attributes in each address.
+    Map<Locality, Integer> localityWeights = new HashMap<>();
+    for (EquivalentAddressGroup eag : resolvedAddresses.getAddresses()) {
+      Attributes eagAttrs = eag.getAttributes();
+      Locality locality = eagAttrs.get(InternalXdsAttributes.ATTR_LOCALITY);
+      Integer localityWeight = eagAttrs.get(InternalXdsAttributes.ATTR_LOCALITY_WEIGHT);
+
+      if (locality == null) {
+        helper.updateBalancingState(TRANSIENT_FAILURE, new ErrorPicker(
+            Status.UNAVAILABLE.withDescription("wrr_locality error: no locality provided")));
+        return false;
+      }
+      if (localityWeight == null) {
+        helper.updateBalancingState(TRANSIENT_FAILURE, new ErrorPicker(
+            Status.UNAVAILABLE.withDescription("wrr_locality error: no locality weight provided")));
+        return false;
+      }
```

Comment:
It seems appropriate to add which locality in  error, no?

Reply:
Good point, I added the locality name.